parquet/src/test/java/org/apache/iceberg/parquet/TestParquetFileMerger.java

```diff
@@ -0,0 +1,156 @@
+    File textFile = new File(tempDir, "not-parquet.txt");
+    java.nio.file.Files.write(textFile.toPath(), "This is not a Parquet file".getBytes());
+
+    InputFile inputFile = Files.localInput(textFile);
+    List<InputFile> inputFiles = Lists.newArrayList(inputFile);
+
+    // Should return false because file is not valid Parquet
+    boolean result = ParquetFileMerger.canMerge(inputFiles);
+    assertThat(result).isFalse();
+  }
+
+  @Test
+  public void testCanMergeReturnsFalseForDifferentSchemas() throws IOException {
+    // Create first Parquet file with schema1
+    Schema icebergSchema1 =
+        new Schema(
+            Types.NestedField.required(1, "id", Types.IntegerType.get()),
+            Types.NestedField.optional(2, "data", Types.StringType.get()));
+
+    File parquetFile1 = new File(tempDir, "file1.parquet");
+    OutputFile outputFile1 = Files.localOutput(parquetFile1);
+    createParquetFileWithSchema(outputFile1, icebergSchema1);
+
+    // Create second Parquet file with different schema
+    Schema icebergSchema2 =
+        new Schema(
+            Types.NestedField.required(1, "id", Types.IntegerType.get()),
+            Types.NestedField.optional(3, "other", Types.LongType.get())); // Different field
+
+    File parquetFile2 = new File(tempDir, "file2.parquet");
+    OutputFile outputFile2 = Files.localOutput(parquetFile2);
+    createParquetFileWithSchema(outputFile2, icebergSchema2);
+
+    // Try to validate - should return false due to different schemas
+    InputFile inputFile1 = Files.localInput(parquetFile1);
+    InputFile inputFile2 = Files.localInput(parquetFile2);
+    List<InputFile> inputFiles = Arrays.asList(inputFile1, inputFile2);
+
+    boolean result = ParquetFileMerger.canMerge(inputFiles);
+    assertThat(result).isFalse();
+  }
+
+  @Test
+  public void testCanMergeReturnsTrueForIdenticalSchemas() throws IOException {
+    // Create two Parquet files with the same schema
+    Schema icebergSchema =
+        new Schema(
+            Types.NestedField.required(1, "id", Types.IntegerType.get()),
+            Types.NestedField.optional(2, "data", Types.StringType.get()));
+
+    File parquetFile1 = new File(tempDir, "file1.parquet");
+    OutputFile outputFile1 = Files.localOutput(parquetFile1);
+    createParquetFileWithSchema(outputFile1, icebergSchema);
+
+    File parquetFile2 = new File(tempDir, "file2.parquet");
+    OutputFile outputFile2 = Files.localOutput(parquetFile2);
+    createParquetFileWithSchema(outputFile2, icebergSchema);
+
+    // Should return true for identical schemas
+    InputFile inputFile1 = Files.localInput(parquetFile1);
+    InputFile inputFile2 = Files.localInput(parquetFile2);
+    List<InputFile> inputFiles = Arrays.asList(inputFile1, inputFile2);
+
+    boolean result = ParquetFileMerger.canMerge(inputFiles);
+    assertThat(result).isTrue();
+  }
+
+  @Test
+  public void testCanMergeReturnsTrueForSingleFile() throws IOException {
```

Comment:
We need to test the row_id related stuff as well

Reply:
Just added